lib/posix/pthread.c

```diff
@@ -597,3 +597,76 @@ int pthread_attr_destroy(pthread_attr_t *attr)
 
 	return EINVAL;
 }
+
+/**
+ * @brief Set name of POSIX thread.
```

Comment:
Sorry for finding more things to comment on, but Zephyr has a pattern of providing docstring in .h files. I'm not even sure if doxygen will pick up docs from source files. Even if does, myself and other developers got used to look up info in .h files, so could you please move docstring there. (Don't get confused by the lack of docstrings to other POSIX(ish) funcs - we kinda assume people could find reference info somewhere else, but eventually would like to add explicit docstrings).